src/figuresClasses.ts

```diff
@@ -1,11 +1,90 @@
-export interface Figure {}
+// export interface Figure {}
 
-export class Triangle implements Figure {}
+// export class Triangle implements Figure {}
 
-export class Circle implements Figure {}
+// export class Circle implements Figure {}
 
-export class Rectangle implements Figure {}
+// export class Rectangle implements Figure {}
 
-export function getInfo(figure): string {
-  return typeof figure;
+// export function getInfo(figure): string {
+//   return typeof figure;
+// }
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('The longest side >= than a sum of two others');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius <= 0');
```

Comment:
The error message 'Radius <= 0' could be more descriptive to provide better context on why the error occurred.